gnc/navigator_perception/scan_the_code_detector.cc

```diff
@@ -193,87 +202,104 @@ void ScanTheCodeDetector::init_ros(stringstream& log_msg){
-    if (true){
-          process_current_image();
+void ScanTheCodeDetector::run()
+{
+    ros::Rate loop_rate(10);  // process images 10 times per second
+    while (ros::ok())
+        {
+            if (true)
+                {
+                    process_current_image();
 
-    }
-    loop_rate.sleep();
-  }
-  return;
+                }
+            loop_rate.sleep();
+        }
+
+    return;
 }
 
-void ScanTheCodeDetector::process_current_image(){
-    Eigen::Vector3d position;
+void ScanTheCodeDetector::process_current_image()
+{
     Mat current_image_left, current_image_right, processing_size_image_left,
         processing_size_image_right;
-
-    try{
-        validate_frame(current_image_left, current_image_right,
-                          processing_size_image_left, processing_size_image_right);
-    }catch(const char* msg){
+    try
+    {
+        validate_frame(current_image_left,
+                          current_image_right,
+                          processing_size_image_left,
+                          processing_size_image_right);
+    }
+    catch(const char* msg)
+    {
+        ROS_ERROR(msg);
         return;
     }
 
-    Matx34d left_cam_mat = this->left_cam_model.fullProjectionMatrix();
+    Matx34d left_cam_mat = left_cam_model.fullProjectionMatrix();
     Matx34d  right_cam_mat = right_cam_model.fullProjectionMatrix();
-
-    bool ret = model_fitter->determine_model_position(position,
-                                                      max_features, feature_block_size,
-                                                      feature_min_distance,
-                                                      image_proc_scale, diffusion_time,
-                                                      current_image_left, current_image_right,
-                                                      processing_size_image_left, processing_size_image_right,
-                                                      left_cam_mat, right_cam_mat);
+    vector<Eigen::Vector3d> position;
+    model_fitter->determine_model_position(position,
+                                           max_features,
+                                           feature_block_size,
+                                           feature_min_distance,
+                                           image_proc_scale,
+                                           diffusion_time,
+                                           current_image_left,
+                                           current_image_right,
+                                           left_cam_mat,
+                                           right_cam_mat);
 }
 
 bool ScanTheCodeDetector::detection_activation_switch(
     sub8_msgs::TBDetectionSwitch::Request &req,
-    sub8_msgs::TBDetectionSwitch::Response &resp) {
+    sub8_msgs::TBDetectionSwitch::Response &resp)
+{
     resp.success = false;
     stringstream ros_log;
     ros_log << "\x1b[1;31mSetting torpedo board detection to: \x1b[1;37m"
```

Comment:
hrmmm, I will fix this when I write my mission and need to start thinking about internode communication. This should be tomorrow